sdk/rust/nym-sdk/src/stream_wrapper/mixnet_stream_wrapper.rs

```diff
@@ -0,0 +1,523 @@
+///
+/// The cause for a MixSocket > going striaght to a MixStream is creating a Nym Client disconnected from the Mixnet first, then upgrading to a Stream when connecting it. Once LP
+/// is implemented, this will also allow us to follow something like what is implemented for the Tokio::net::UdpFramed abstraction, where we can create multiple MixStream instances
+/// from a single MixSocket, all connected to different Recipients (aka a one-to-many setup with a single MixnetClient).
+pub struct MixSocket {
+    pub inner: MixnetClient,
+}
+
+impl MixSocket {
+    /// Create a new socket that is disconnected from the Mixnet - kick off the Mixnet client with config for builder.
+    /// Following idea of having single client with multiple concurrent connections represented by per-Recipient MixStream instance.
+    pub async fn new() -> Result<Self, Error> {
+        // TODO make this take an option, if Some create with builder pattern, if None make ephemeral like we already are.
+        let inner = MixnetClient::connect_new().await?;
+        Ok(MixSocket { inner })
+    }
+
+    /// Connect to a specific peer (Nym Client) and return a Stream (cf TcpSocket::connect() / TcpStream::new()).
+    pub async fn connect_to(_recipient: Recipient) -> Result<MixStream, Error> {
+        todo!()
+    }
+
+    /// Get our Nym address.
+    pub fn nym_address(&self) -> &Recipient {
+        self.inner.nym_address()
+    }
+
+    pub fn get_ref(&self) -> &MixnetClient {
+        &self.inner
+    }
+
+    pub fn get_mut(&mut self) -> &mut MixnetClient {
+        &mut self.inner
+    }
+
+    pub fn into_inner(self) -> MixnetClient {
+        self.inner
+    }
+}
+
+pub struct MixStream {
+    pub client: MixnetClient,
+    peer: Option<Recipient>, // We might be accepting incoming messages and replying, so might not have a Nym addr to talk to..
+    peer_surbs: Option<AnonymousSenderTag>, // ..since we might just be using SURBs instead
+}
+
+impl MixStream {
+    /// Create a MixStream instance and immediately connect (convenience method) or pass in a MixSocket (pre-configured DisconnectedMixnetClient) and connect it.
+    // TODO in future take config from MixSocket if exists in Option<> param, else spin up ephemeral client. Just doing ephemeral for initial sketch.
+    pub async fn new(socket: Option<MixSocket>, peer: Recipient) -> Self {
+        let client = match socket {
+            Some(socket) => socket.into_inner(),
+            None => MixnetClient::connect_new().await.unwrap(),
+        };
+        Self {
+            client,
+            peer: Some(peer),
+            peer_surbs: None,
+        }
+    }
+
+    /// Nym address of Stream's peer (Nym Client it will communicate with).
+    pub fn peer_addr(&self) -> Recipient {
+        let peer = &self.peer.expect("No Peer set");
+        *peer
+    }
+
+    /// Our Nym address.
+    pub fn local_addr(&self) -> &Recipient {
+        self.client.nym_address()
+    }
+
+    /// Store SURBs in own field.
```

Comment:
does it though? you just provide it a tag. no actual surbs are attached

Reply:
Yes this is a vague comment, its more storing the tag to use for anonymous replies, will change